test/profile_eospac.cpp

```diff
@@ -0,0 +1,360 @@
+// © 2021-2022. Triad National Security, LLC. All rights reserved.  This
+// program was produced under U.S. Government contract 89233218CNA000001
+// for Los Alamos National Laboratory (LANL), which is operated by Triad
+// National Security, LLC for the U.S.  Department of Energy/National
+// Nuclear Security Administration. All rights in the program are
+// reserved by Triad National Security, LLC, and the U.S. Department of
+// Energy/National Nuclear Security Administration. The Government is
+// granted for itself and others acting on its behalf a nonexclusive,
+// paid-up, irrevocable worldwide license in this material to reproduce,
+// prepare derivative works, distribute copies to the public, perform
+// publicly and display publicly, and to permit others to do so.
+//------------------------------------------------------------------------------
+
+/*
+  Profiling tool for EOSPAC EOS
+  Authors: Jonah Miller, Chad Meyers, Josh Dolence, Daniel Holiday,
+           Richard Berger
+  Copyright: LANL
+ */
+
+#include <chrono>
+#include <cmath>
+#include <cstdlib>
+#include <fstream>
+#include <iomanip>
+#include <iostream>
+#include <limits>
+#include <random>
+#include <string>
+
+#include <ports-of-call/portability.hpp>
+#include <ports-of-call/portable_arrays.hpp>
+#include <spiner/databox.hpp>
+#include <spiner/interpolation.hpp>
+
+#include <singularity-eos/eos/eos.hpp>
+#include <singularity-eos/eos/eos_builder.hpp>
+
+using namespace singularity;
+
+using duration = std::chrono::microseconds;
+using dvec = std::vector<double>;
+using ivec = std::vector<int>;
+using Spiner::DataBox;
+using Spiner::RegularGrid1D;
+
+#ifdef PORTABILITY_STRATEGY_KOKKOS
+using RView = Kokkos::View<Real *>;
+using RMirror = typename RView::HostMirror;
+#endif
+
+constexpr Real RHO_MIN = 1e-2; // g/cm^3
+constexpr Real RHO_MAX = 1e2;
+constexpr Real T_MIN = 1e2; // Kelvin
+constexpr Real T_MAX = 1e4;
+constexpr Real E_MIN = 1e9;
+constexpr Real E_MAX = 1e13;
+
+inline auto now() { return std::chrono::high_resolution_clock::now(); }
+
+template <typename Function>
+inline double get_duration(Function function) {
+  auto start = now();
+  function();
+  auto stop = now();
+  return std::chrono::duration_cast<duration>(stop - start).count();
+}
+
+//#define METHOD_UNDER_TEST TemperatureFromDensityInternalEnergy
+//#define METHOD_UNDER_TEST PressureFromDensityTemperature
+//#define METHOD_UNDER_TEST InternalEnergyFromDensityTemperature
+//#define METHOD_UNDER_TEST PressureFromDensityInternalEnergy
+//#define METHOD_UNDER_TEST SpecificHeatFromDensityTemperature
+//#define METHOD_UNDER_TEST SpecificHeatFromDensityInternalEnergy
+//#define METHOD_UNDER_TEST BulkModulusFromDensityTemperature
+//#define METHOD_UNDER_TEST BulkModulusFromDensityInternalEnergy
+//#define METHOD_UNDER_TEST GruneisenParamFromDensityTemperature
+//#define METHOD_UNDER_TEST GruneisenParamFromDensityInternalEnergy
```

Comment:
Were these for debugging or do we need to leave them in?

Reply:
It was left as a comment to document how to test different methods. The CMake logic generates executables with each one of these defines. I have no preferences either way.

Reply:
We can leave them, but can you add a comment above to just explain what's going on here?

Reply:
Added a comment above.